### DISCUSSION TITLE
incomplete data at data.nasdaq.com for historic tickers

### DISCUSSION THREAD

**Discussion by jeejee19** [fja05680/sp500#11] [Q&A]
Hi, really appreciate the work you have done in building and maintaining these data. I am using the tickers to pull data from [https://data.nasdaq.com/api/v3/datatables/QUOTEMEDIA/PRICES?date=2007-05-02&ticker=abi&api_key="APIKEY](https://data.nasdaq.com/api/v3/datatables/QUOTEMEDIA/PRICES?date=2007-05-02&ticker=abi&api_key=%22APIKEY)". In this example I have included ABI as a ticker in 2007 (per your dataset), however the nasdaq link is empty. This happens for some 100 tickers in 2007. Another 400 are ok. Would you happen to know if I need to do a ticker conversion of some type? Thanks!

PS. seems ticker RTVY is missing in your data (according to wikipedia in since 1985).

**Comment by fja05680** [comment]
jeejee19,

Thanks for your kind words.  Do you have paid or free subscription to that provider?

The ticker symbols should be as they were in 2007, in this case.  Unfortunately, I have no way to independently check using any free source.  Sorry I know this isn't helpful.

For RTVY, can you send me the wikipedia link?  I couldn't find it.

Farrell

**Reply by jeejee19** [reply]
Hi Farrell, 

Indeed a paid subscription. I am surprised this paid subscription is incomplete. Just to check, in the sources you apply your set of tickers, you do get NYSE EOD data for ABI in 2007? I checked historic pages on Wikipedia and indeed ABI appears here:
https://en.wikipedia.org/w/index.php?title=List_of_S%26P_500_companies&oldid=130959051
(Applera Corp-Applied Biosystems Group)

----

On RTVY
This is the wikipedia link: https://en.wikipedia.org/wiki/List_of_S%26P_500_companies

Copied details below.
RVTY | Revvity | Health Care | Health Care Equipment | Waltham, Massachusetts | 1985-05-31 | 0000031791 | 1937
-- | -- | -- | -- | -- | -- | -- | --
[RVTY](https://www.nyse.com/quote/XNYS:RVTY)	[Revvity](https://en.wikipedia.org/wiki/Revvity)	Health Care	Health Care Equipment	[Waltham, Massachusetts](https://en.wikipedia.org/wiki/Waltham,_Massachusetts)	1985-05-31	0000031791	1937

Kind regards, Jan-Jaap

**Reply by fja05680** [reply]
Jan-Jaap,

On May 16, 2023, PKI symbol was changed to RVTY, see:
https://www.marketscreener.com/quote/stock/REVVITY-INC-12426/news/Revvity-Inc-will-Change-its-Ticker-to-RVTY-from-PKI-43862811/

In my next update, I'll add an entry for May 19, 2023 that will remove the PKI and add RVTY.

I went back to the original csv file 'S&P 500 Historical Components & Changes.csv' and noticed that ABI appears as ABI-200811.  After a symbol is removed,  a suffix is appended to the end of the symbol that uses a dash and the removal date.  I strip these dates out in my updated list ('S&P 500 Historical Components & Changes(04-16-2023).csv').  You might try to see if you can retrieve `ABI-200811`.

In the book "Trading Evolved" where I got the original list, Andreas says he used "Norgate Data" for historical stock data.  I've personally never tried to back test equities.  I've just stuck to ETFs to avoid a lot of the complexities.  Shameless plug - my backtester "Pinkfish"  is really geared toward that.

If it turns out that the end-date suffix is what is needed, I could produce another list that has those in there.

Also, it's interesting that you can get some historical lists from wikipedia itself.  Thanks for pointing that out.  It makes me wonder how much of the historical list could be reconstructed this way.

Farrell

**Reply by jeejee19** [reply]
Ah RTVY indeed very new.. 
I tried the option with suffix, sofar no success. Will have a look to Norgate too and it triggers some more ideas with me. Anyway, I sent a mail to helpdesk data.nasdaq.com, let's see their response.
And will have a look to Pinkfish :)
